Library/Homebrew/test/test_versions.rb

```diff
@@ -54,8 +54,10 @@ def test_comparing_regular_versions
   end
 
   def test_HEAD
-    assert_operator version("HEAD"), :>, version("1.2.3")
-    assert_operator version("1.2.3"), :<, version("HEAD")
+    assert_operator head_version("HEAD"), :>, version("1.2.3")
+    assert_operator head_version("HEAD-abcdef"), :>, version("1.2.3")
```

Comment:
The idea with `HEAD` versions is that if there is some upstream version that differs from one of installed `HEAD`s, then all of installed `HEAD`s are outdated. So we can compare `HEAD`-versions, but not using `HeadVersion#<=>` because to compare `HEAD`-versions inside `HeadVersion` we need more than just their versions strings, we need add `downloader` or `formula` to `HeadVersion`. 

> It might also not be a bad idea to mandate a version if e.g. git describe --tags doesn't return anything so we can compare between HEAD versions and normal versions (but obviously that doesn't need to be part of this PR).

In the original PR there is some logic added to the `Tab` which can allow us to compare `HEAD` versions with normal versions: tabs store latest `stable` and `devel` versions at the moment the keg is installed, so if `HEAD` keg's  `stable` is less than current stable version, then `HEAD`-keg is outdated.